### PR TITLE
[squeezebox] Cleanup / code simplification

### PR DESCRIPTION
**Pull request by maniac103**
Improve code readability a bit by
- using lambdas for anonymous listener instantiations
- decoding URL encoded responses earlier so the code can check against non-encoded responses

### CODE REVIEW

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java [hunk 1]
```diff
@@ -997,25 +908,32 @@ private void handleFavorites(String message) {
...
             boolean isTypePlaylist = false;
             for (String part : messageParts) {
+                final String decoded = decode(part);
+                int colonPos = decoded.indexOf(":");
+                if (colonPos < 0) {
+                    continue;
+                }
+
+                final String key = decoded.substring(0, colonPos);
+                final String value = decoded.substring(colonPos);
```
Comment (mhilbush) [inline bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java:918]: Shouldn't this be `colonPos + 1`?

**Review by mhilbush — CHANGES_REQUESTED** [review]
I left one comment for you to check.

Otherwise, LGTM

File: bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java (hunk 1 above)
Comment (maniac103) [reply bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java]: Oops, of course. Thanks, fixed.

Comment (maniac103) [reply bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java]: NB: I thought about factoring out this parsing to a separate method, but due the lack of a destructuring binding in Java (e.g. Kotlin: `val (key, value) = ...` this didn't really improve readability.

Comment (maniac103) [reply bundles/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/handler/SqueezeBoxServerHandler.java]: Ended up going a step back and parsed the full CLI response into a map.

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**